tests/engine/training/test_fingerprinting.py

```diff
@@ -76,3 +83,44 @@ def test_fingerprint_changes_due_to_changed_source(monkeypatch: MonkeyPatch):
     assert key1 != key2
 
     get_source_mock.assert_called_once_with(TEDPolicy)
+
+
+def test_fingerprint_changes_when_external_file_changes():
```

Comment:
OPT: can you split this into two tests: one testing that the fingerprint is stable and another to check it is changing?